app/api/chemotion/report_api.rb

```diff
@@ -198,5 +198,146 @@ class ReportAPI < Grape::API
         excel.generate_file
       end
     end
+
+    resource :multiple_reports do
+      desc "Build a multi-reactions report using the contents of a JSON file"
+
+      params do
+        requires :ids
+        requires :settings
+      end
+
+      get :rtf do
+        ids = params[:ids].split("_")
+        settings = params[:settings].split("_")
+
+        report = Report::RTFReport.new do |r|
```

Comment:
Looks not good. Please make a module for that. Example

https://github.com/ComPlat/chemotion_ELN/blob/master/lib/svg/reaction_composer.rb

You can do this without a class, just a module with function would also fine. However if you want you can build a class like done in the example.

In general, Rails and API actions have to laconic and don't contains much logic.